src/main/java/com/programmers/library/service/LibraryService.java

```diff
@@ -0,0 +1,147 @@
+import com.programmers.library.utils.StatusType;
+import com.programmers.library.repository.LibraryRepository;
+import com.programmers.library.utils.MessageType;
+
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class LibraryService {
+
+    private LibraryRepository libraryRepository;
+
+    public LibraryService(LibraryRepository libraryRepository) {
+        this.libraryRepository = libraryRepository;
+    }
+
+    // 메시지 출력
+    private void printMessage(String message, String status) {
+        System.out.println("[System] " + message + status);
+    }
+
+    // 도서 등록
+    public void addBook(String title, String author, int pages) {
+        Book book = new Book(title, author, pages);
+        libraryRepository.save(book);
+        printMessage(MessageType.ADD_SUCCESS.getMessage(), "\n");
+    }
+
+    // 전체 도서 목록 조회
+    public void viewAllBooks() {
+        List<Book> books = libraryRepository.findAll();
+
+        if(books.isEmpty()) {   // 현재 등록된 도서가 없는 경우
+            printMessage(MessageType.NO_BOOKS.getMessage(), "\n");
+            return;
+        }
+
+        printMessage(MessageType.LIST_START.getMessage(), "\n");
+        for (Book book : books) {
+            System.out.println(book.toString());
+            System.out.println("\n------------------------------\n");
+        }
+        printMessage(MessageType.LIST_END.getMessage(), "\n");
+    }
+
+    // 도서 제목으로 검색
+    public void searchBookByTitle(String title) {
+        List<Book> books = libraryRepository.findByTitle(title);
+
+        if(books.isEmpty()) {
+            printMessage(MessageType.NO_SEARCH_BOOKS.getMessage(), "\n");
+            return;
+        }
+
+        for (Book book : books) {
+            System.out.println("\n" + book.toString());
+            System.out.println("\n------------------------------");
+        }
+        printMessage(MessageType.SEARCH_END.getMessage(), "\n");
+    }
+
+    // 도서 대여
+    public void rentBook(int bookId) {
+        libraryRepository.findById(bookId)
+                .ifPresentOrElse(
+                        book -> {
+                            StatusType status = book.getStatus();
+
+                            switch (status) {
+                                case AVAILABLE -> {
+                                    libraryRepository.updateStatus(bookId, StatusType.RENTING);
+                                    printMessage(MessageType.RENT_SUCCESS.getMessage(), "\n");
+                                }
+                                case RENTING -> printMessage(MessageType.ALREADY_RENTED.getMessage(), "\n");
+                                case ORGANIZING, LOST -> printMessage(MessageType.NOT_AVAILABLE.getMessage(), "( *사유: " + status.getDescription() + " )\n");
+                            }
+                        },
+                        () -> printMessage(MessageType.BOOK_NOT_FOUND.getMessage(), "\n")
+                );
+    }
```

Comment:
```suggestion
    public void rentBook(int bookId) {
        libraryRepository.findById(bookId)
                .ifPresentOrElse(
                        book -> updatedStatus(bookId, book),
                        () -> printMessage(MessageType.BOOK_NOT_FOUND.getMessage(), "\n")
                );
    }

    private void updatedStatus(int bookId, Book book) {
        StatusType status = book.getStatus();

        switch (status) {
            case AVAILABLE -> {
                libraryRepository.updateStatus(bookId, StatusType.RENTING);
                printMessage(MessageType.RENT_SUCCESS.getMessage(), "\n");
            }
            case RENTING -> printMessage(MessageType.ALREADY_RENTED.getMessage(), "\n");
            case ORGANIZING, LOST -> printMessage(MessageType.NOT_AVAILABLE.getMessage(), "( *사유: " + status.getDescription() + " )\n");
        }
    }
```

depth가 깊어지면 코드를 분석하거나 리펙토링하기 어렵습니다.
우선 메서드를 분리하는 것부터 시작해보시죠!